packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -78,18 +76,23 @@ function outerHTMLGetterPatched(this: Element) {
     return getOuterHTML(this);
 }
 
-function attachShadowPatched(this: Element, options: ShadowRootInit): SyntheticShadowRootInterface {
-    return attachShadow(this, options);
+function attachShadowPatched(this: Element, options: ShadowRootInit): ShadowRoot {
+    // To retain native behavior of the API, provide synthetic shadowRoot only when specified
+    if (isTrue(options['$$lwc-synthetic-mode$$']) || isUndefined(originalAttachShadow)) {
+        return attachShadow(this, options);
+    } else {
+        return originalAttachShadow.call(this, options);
```

Comment:
I think we should have another if inside this block with a more descriptive error saying that you're running LWC in an environment that doesn't support shadow and you're missing the synthetic-shadow polyfill.

Reply:
Can you explain a bit more? 
This code is in the synthetic-shadow, so it can be assumed that the polyfill is loaded.